src/components/ComplaintDetail/ComplaintDetail.js

```diff
@@ -0,0 +1,62 @@
+import './ComplaintDetail.less';
+import React, { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  selectDetailActiveCall,
+  selectDetailData,
+  selectDetailError,
+} from '../../reducers/detail/selectors';
+import { Link, useLocation, useParams } from 'react-router-dom';
+import { getComplaintDetail } from '../../actions/complaints';
+import iconMap from '../iconMap';
+import Loading from '../Dialogs/Loading';
+import { selectQuerySearch } from '../../reducers/query/selectors';
+import { ComplaintDetailBody } from './ComplaintDetailBody';
+
+export const ComplaintDetail = () => {
+  const location = useLocation();
+  const { id } = useParams();
+  const dispatch = useDispatch();
+  const data = useSelector(selectDetailData);
+  const error = useSelector(selectDetailError);
+  const activeCall = useSelector(selectDetailActiveCall);
+  const search = useSelector(selectQuerySearch);
+
+  const isLoading = activeCall !== '';
+  const backUrl = useMemo(() => {
+    const pathName = location.pathname || '';
+    const idx = pathName.indexOf('detail');
```

Comment:
If `pathName` ends up as '', `idx` will end up as `''.indexOf('detail')`, which will equal `-1`. Is it a problem if `idx` equals `-1`?

Reply:
Or, if `pathName` would never end up in that state, we should strip `|| ''` from the prior line.

Reply:
The || '' is for when the page first loads that the object isn't ready it.

It seemed to bomb on initial load.

I guess I can just return an empty string and exit if location isn't ready yet.